hugolib/author.go

```diff
@@ -43,3 +78,205 @@ type Author struct {
 // - linkedin
 // - skype
 type AuthorSocial map[string]string
+
+// URL is a convenience function that provides the correct canonical URL
+// for a specific social network given a username. If an unsupported network
+// is requested, only the username is returned
+func (as AuthorSocial) URL(key string) string {
+	switch key {
+	case "github":
+		return fmt.Sprintf("https://github.com/%s", as[key])
+	case "facebook":
+		return fmt.Sprintf("https://www.facebook.com/%s", as[key])
+	case "twitter":
+		return fmt.Sprintf("https://twitter.com/%s", as[key])
+	case "googleplus":
+		isNumeric := onlyNumbersRegExp.Match([]byte(as[key]))
+		if isNumeric {
+			return fmt.Sprintf("https://plus.google.com/%s", as[key])
+		}
+		return fmt.Sprintf("https://plus.google.com/+%s", as[key])
+	case "pinterest":
+		return fmt.Sprintf("https://www.pinterest.com/%s/", as[key])
+	case "instagram":
+		return fmt.Sprintf("https://www.instagram.com/%s/", as[key])
+	case "youtube":
+		return fmt.Sprintf("https://www.youtube.com/user/%s", as[key])
+	case "linkedin":
+		return fmt.Sprintf("https://www.linkedin.com/in/%s", as[key])
+	default:
+		return as[key]
+	}
+}
+
+func mapToAuthors(m map[string]interface{}) Authors {
+	authors := make(Authors, 0, len(m))
+	for authorID, data := range m {
+		authorMap, ok := data.(map[string]interface{})
+		if !ok {
+			continue
+		}
+		a := mapToAuthor(authorID, authorMap)
+		if a.ID != "" {
+			authors = append(authors, a)
+		}
+	}
+	sort.Stable(authors)
+	return authors
+}
+
+func mapToAuthor(id string, m map[string]interface{}) Author {
+	if id == "" {
+		return Author{}
+	}
+
+	author := Author{ID: id}
+	for k, data := range m {
+		switch strings.ToLower(k) {
+		case "givenname", "firstname":
+			author.GivenName = cast.ToString(data)
+			author.FirstName = author.GivenName
+		case "familyname", "lastname":
+			author.FamilyName = cast.ToString(data)
+			author.LastName = author.FamilyName
+		case "displayname":
+			author.DisplayName = cast.ToString(data)
+		case "thumbnail":
+			author.Thumbnail = cast.ToString(data)
+		case "image":
+			author.Image = cast.ToString(data)
+		case "shortbio":
+			author.ShortBio = cast.ToString(data)
+		case "bio":
+			author.Bio = cast.ToString(data)
+		case "email":
+			author.Email = cast.ToString(data)
+		case "weight":
+			author.Weight = cast.ToInt(data)
+		case "social":
+			author.Social = normalizeSocial(cast.ToStringMapString(data))
+		case "params":
+			author.Params = cast.ToStringMapString(data)
+		case "languages":
+			if author.languages == nil {
```

Comment:
Authors are only defined in `/data/_authors` and the author key/ID is referenced in front matter.

If you don't have any specific suggestions, I will just leave the implementation as is, which works, is clear and is tested.